### PR TITLE
KTOR-3721 KTOR-3707 Status pages Fixes

### PR DESCRIPTION
**Pull request by e5l**
Verify [KTOR-3721](https://youtrack.jetbrains.com/issue/KTOR-3721)
Verify [KTOR-3707](https://youtrack.jetbrains.com/issue/KTOR-3707)

### CODE REVIEW

File: ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/routing/Routing.kt [hunk 1]
```diff
@@ -41,12 +40,15 @@ public class Routing(
...
-        when (val resolveResult = resolveContext.resolve()) {
+        val resolveResult = resolveContext.resolve()
+        when (resolveResult) {
             is RoutingResolveResult.Success ->
                 executeResult(context, resolveResult.route, resolveResult.parameters)
             is RoutingResolveResult.Failure ->
                 context.call.attributes.put(RoutingFailureStatusCode, resolveResult.errorStatusCode)
         }
+
+        println("Resolve finished: $resolveResult")
```
Comment (rsinukov) [inline ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/routing/Routing.kt:51]: Please remove `println`

File: ktor-server/ktor-server-plugins/ktor-server-status-pages/jvmAndNix/src/io/ktor/server/plugins/statuspages/StatusPages.kt [hunk 2]
```diff
@@ -9,11 +9,16 @@ import io.ktor.http.content.*
 import io.ktor.server.application.*
 import io.ktor.util.*
 import io.ktor.util.reflect.*
-import kotlin.native.concurrent.*
 import kotlin.reflect.*
 
+/**
+ * Specify how the exception should be handled.
```
Comment (rsinukov) [inline ktor-server/ktor-server-plugins/ktor-server-status-pages/jvmAndNix/src/io/ktor/server/plugins/statuspages/StatusPages.kt:15]: `Specifies`

File: ktor-server/ktor-server-plugins/ktor-server-status-pages/jvmAndNix/test/io/ktor/server/plugins/statuspages/StatusPageTest.kt [hunk 3]
```diff
@@ -381,4 +369,71 @@ class StatusPageTest {
             assertEquals("NotFound", call.response.content)
         }
     }
+
+    @Test
+    fun testStatusPagesNotCalledAfterRouting() {
```
Comment (rsinukov) [inline ktor-server/ktor-server-plugins/ktor-server-status-pages/jvmAndNix/test/io/ktor/server/plugins/statuspages/StatusPageTest.kt:374]: `testRoutingNotCalledAfterStatusPages`

**Review by rsinukov — CHANGES_REQUESTED** [review]
Please check CI build

File: ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/routing/Routing.kt (hunk 1 above)
Comment (e5l) [reply ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/routing/Routing.kt:51]: Thanks, will do

File: ktor-server/ktor-server-plugins/ktor-server-status-pages/jvmAndNix/test/io/ktor/server/plugins/statuspages/StatusPageTest.kt (hunk 3 above)
Comment (e5l) [reply ktor-server/ktor-server-plugins/ktor-server-status-pages/jvmAndNix/test/io/ktor/server/plugins/statuspages/StatusPageTest.kt:374]: Exactly, will rename

File: ktor-server/ktor-server-plugins/ktor-server-status-pages/jvmAndNix/test/io/ktor/server/plugins/statuspages/StatusPagesTest.kt [hunk 4]
```diff
@@ -0,0 +1,462 @@
...
+            client.get("/").let { response ->
+                assertEquals(HttpStatusCode.InternalServerError, response.status)
+                assertEquals("IllegalStateException", response.bodyAsText())
+            }
+        }
+    }
+
+    @Test
+    fun testErrorShouldNotRecurse() {
+        testApplication {
```
Comment (rsinukov) [inline ktor-server/ktor-server-plugins/ktor-server-status-pages/jvmAndNix/test/io/ktor/server/plugins/statuspages/StatusPagesTest.kt:213]: please use `testErrorShouldNotRecurse() = testApplication` syntax

**Review by rsinukov — APPROVED** [review]

**Pull request merged**